### PR TITLE
Adds description parameter to TFE Workspace

### PR DESCRIPTION
**Pull request by petems**
* Requires go-tfe update

## Description

Adds description to workspace 

## External links

_Include any links here that might be helpful for people reviewing your PR. If there are none, feel free to delete this section._

- https://github.com/hashicorp/go-tfe/issues/170

https://www.terraform.io/docs/cloud/api/workspaces.html#request-body

data.attributes.description | string | (nothing) | A description for the workspace.
-- | -- | -- | --

Required in create and read

### CODE REVIEW

File: tfe/resource_tfe_workspace.go [hunk 1]
```diff
@@ -46,6 +46,12 @@ func resourceTFEWorkspace() *schema.Resource {
 				ForceNew: true,
 			},
 
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  false,
```
Comment (chrisarcand) [inline tfe/resource_tfe_workspace.go:52]: `false`? I think you want `""` here.

**Review by chrisarcand — CHANGES_REQUESTED** [review]
* Needs some test additions 💞 
* Run a `go mod tidy` to remove the 0.12.0 refs, too.

File: tfe/resource_tfe_workspace.go (hunk 1 above)
Comment (petems) [reply tfe/resource_tfe_workspace.go]: Fixed! 👍

**Comment by petems** [comment]
@chrisarcand Added tests 🖖

File: tfe/data_source_workspace_test.go [hunk 2]
```diff
@@ -63,6 +65,7 @@ resource "tfe_organization" "foobar" {
 resource "tfe_workspace" "foobar" {
   name                  = "workspace-test-%d"
   organization          = tfe_organization.foobar.id
+  description						= "provider-testing"
```
Comment (chrisarcand) [inline tfe/data_source_workspace_test.go:68]:
```suggestion
  description           = "provider-testing"
```

File: tfe/resource_tfe_workspace.go [hunk 3]
```diff
@@ -46,6 +46,12 @@ func resourceTFEWorkspace() *schema.Resource {
 				ForceNew: true,
 			},
 
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "",
```
Comment (chrisarcand) [inline tfe/resource_tfe_workspace.go:52]: Sorry, I know we already changed this from `false` before....I don't believe this is necessary at all, right? Descriptions can be `null`; the zero value of the field here will naturally be `""` anyway.

File: tfe/resource_tfe_workspace_test.go [hunk 4]
```diff
@@ -53,6 +53,47 @@ func TestAccTFEWorkspace_basic(t *testing.T) {
...
+					resource.TestCheckResourceAttr(
+						"tfe_workspace.foobar", "trigger_prefixes.#", "0"),
+					resource.TestCheckResourceAttr(
+						"tfe_workspace.foobar", "working_directory", ""),
+				),
+			},
+		},
+	})
+}
+
```
Comment (chrisarcand) [inline tfe/resource_tfe_workspace_test.go:96]: I don't think this simple field merits a separate expensive test; let's move it in to `TestAccTFEWorkspace_basic` with other basic arguments.

**Review by chrisarcand — CHANGES_REQUESTED** [review]
Hey! Sorry for the delay.

**Review by chrisarcand — APPROVED** [review]

**Pull request merged**